gapic/templates/%namespace/%name_%version/%sub/services/%service/client.py.j2

```diff
@@ -552,16 +564,52 @@ class {{ service.client_name }}(metaclass={{ service.client_name }}Meta):
                 self._use_mtls_endpoint))
 
         if not transport_provided:
+            {# TODO(https://github.com/googleapis/gapic-generator-python/issues/2121): Remove this condition when async rest is GA. #}
+            {% if rest_async_io_enabled %}
+            transport_init: Union[Type[{{ service.name  }}Transport], Callable[..., {{ service.name }}Transport]] = (
+                {{ service.client_name  }}.get_transport_class(transport)
+                if isinstance(transport, str) or transport is None
+                else cast(Callable[..., {{ service.name }}Transport], transport)
+            )
+
+            if "rest_asyncio" in str(transport_init):
+                # Note: The following parameters are not support in async rest:
+                # credentials_file, scopes, client_cert_source_for_mtls, quota_project_id
+                # always_use_jwt_access, api_audience.
```

Comment:
```suggestion
                # TODO(<bug>): The following parameters are not supported in async rest:
                # `credentials_file, scopes`, `client_cert_source_for_mtls`, `quota_project_id`
                # `always_use_jwt_access`, `api_audience`.
```

Reply:
There's no need to list the parameters in the comment when they're listed very clearly in the code just below.

```suggestion
                # TODO(<bug>): Support the following parameters in async rest:
```

Reply:
addressed.